### ISSUE TITLE
Folder /.actors/ are scraped as movie "The Actors"

### ISSUE THREAD

**Issue by rainman74** [xbmc/xbmc#18255]
### Describe the bug
Here is a clear and concise description of what the problem is:

The movie "The Actors" is often additionally created during scraping if there is a ".actors" folder in the directory.

This film is additionally created for the different films.

Example:
smb:/Folder/Film1/Film1.mkv -> additionally the movie "The Actors" is found and written to the DB with this path smb:/Folder/Film1/.actors/VIDEO_TS.IFO

smb:/Folder/Film2/Film2.mkv -> additionally the movie "The Actors" is found and written to the DB with this path smb:/Folder/Film2/.actors/VIDEO_TS.IFO

The file VIDEO_TS.IFO is not in any of the folders!

Please look at screenshots, too!

### To Reproduce

Start scraping for new movies

### Debuglog
The debuglog can be found here:

not available anymore

### Screenshots 
Here are some links or screenshots to help explain the problem:

https://i.imgur.com/Ec4cgP7.png
https://i.imgur.com/I28KYa8.png

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

not necessary!

### Your Environment
Used Operating system:

Android TV

Operating system version/name:

Kodi version: 19 Alpha 1

**Comment by DaVukovic** [comment]
Well, the log is for sure necessary. Feel free to not provide it. But be aware, that we will simply close this issue if no log is provided.

**Comment by rainman74** [comment]
I don't have the log anymore because I'm already back to Kodi 18.8.

But I have refined the description and also added 2 screenshots.

I already had this problem several months ago with Kodi 19, and I thought it would be fixed by now.

With the Movie Scraper something goes wrong in Kodi 19.

**Comment by rainman74** [comment]
There are more problems when scraping movies, e.g. after converting videoxx.db from Kodi 18.8 to 19 and then searching for new movies, a lot of the already existing movies are found again, even though they were all correctly listed in the DB with the Kodi 18 version of the database!

**Comment by KarellenX** [comment]
I have no problem with the .actors folder for movies and tv shows and I have thousands of them and never experienced your problem.

Without a debuglog, we cannot help you solve your issue. It is not a Kodi bug, but a problem with your setup.

Sure you don't have your log anymore, so why does that prevent you from creating a new one? Delete the movie from the library and rescrape it. Simple. 

Your issue will be closed as it is not reproducible and Kodi cannot invent files that are not there. If it does turn out to be a bug, we will reopen it.

**Comment by rainman74** [comment]
Exactly, it's a bug with my setup! Sure :-)

Everthing is all right with Kodi 18.8 and after converting the VideoXX.db for Kodi 19 it becomes a complete mess.

Well, you have to know what you do with it...

**Comment by KarellenX** [comment]
Final time. Please provide the debug log.

**Comment by rainman74** [comment]
Like I said, I don't have that version anymore. I've already tried to test Kodi 19 twice now and the setup with addons, converting, etc. took an hour each time and then I have to roll everything back to Kodi 18.8.

There is something wrong with converting VideoXX.db, I just wanted to tell you.

I understand that you want the debug.log, but would it even record exact details of the VideoXX.db conversion process? This happens automatically the first time you start Kodi 19.

**Comment by rainman74** [comment]
I've reproduced it twice already. Just ignore it and be happy when the users of Kodi 19 beta or final will migrate the video database.

Just tell me if the db migration steps and errors are written to the debug log or not! If yes, then I might do it again...

**Comment by KarellenX** [comment]
Your report makes no sense.

In your very first post you stated that **when you scrape, the actors folder is being scraped as a movie**

Your report has now changed to some issue when upgrading to v19.

I don't understand what problem you are reporting. Upgrading to v19 does not scrape your library, so how do the two issues connect?

It is great that you are reproducing it as it definitely shows, at the very least, a problem with your setup. But who knows what that problem is.

And you do realise, that while I am responding, your issue is not being looked at. PROVIDE THE DEBUG LOG and stop being insulting.

**Comment by DaVukovic** [comment]
>Upgrading to v19 does not scrape your library, so how do the two issues connect

Probably because the upgrading process converts the content from MyVideos117 to a matching database schema for MyVideos118. Maybe something goes wrong during that conversion. I neither can tell not am I able to reproduce as I can't install Kodi 18 currently to test with

**Comment by KarellenX** [comment]
@DaVukovic 
That is not what is stated in his original post. It clearly states that when he scrapes, the actors folder is also scraped. He has then later thrown in an issue about upgrading. How could it be possible that a database upgrade will add new movies to the database and link those movies to actors images. 

The issue is the OP is unhelpful and refuses to provide the required information. 

This is my last post on this issue. I don't have time for these games.

**Comment by the-black-eagle** [comment]
@DaVukovic If you have a 'MyVideos117' as well as 'MyVideos118' just delete the later version and then start Matrix.  It'll find the older db and upgrade it. FWIW, I can't reproduce either of the issues here.  Upgrading works fine and my actors directory is not scraped when adding new stuff to v19.

**Comment by Dnkhatri** [comment]
> @DaVukovic If you have a 'MyVideos117' as well as 'MyVideos118' just delete the later version and then start Matrix. It'll find the older db and upgrade it. FWIW, I can't reproduce either of the issues here. Upgrading works fine and my actors directory is not scraped when adding new stuff to v19.

kodi 18 is MyVideos116 not MyVideos117 though. I am using Android  didn't see this either though I went MyVideos116>117>118 maybe this happens  when its updated directly from 116>118. I cant test this though as I already dropped 116

**Comment by rainman74** [comment]
I describe the error in more detail above...

**Comment by rainman74** [comment]
I have described the error in more detail above.

And please excuse the title of the bug, I'm afraid I can't tell exactly if the error happens already when converting the MyVideos116 (which I suspect) or if it's only caused by scraping. So I decided to describe the scraped first, but then I adjusted it again.

I hope the description might help more now.

**Comment by fuzzard** [comment]
Provide a log, how many times does it need to be said

**Comment by rainman74** [comment]
> Provide a log, how many times does it need to be said

I have always attached logs to many error messages in the past and still nobody has taken care of it.

If somebody confirms that the conversion process of myVideosXX.db is also visible in it and the log helps in this case, I might do it again.

**Comment by DaVukovic** [comment]
I just tried on an older Ubuntu installation I had lying around (18.04) and I did the Kodi update from 18.8 to 19 (MyVideos116  -> 118 ...and you are for sure right @Dnkhatri  mixed that up ;) ) and I also couldn't reproduce

```
- Now I start a new scraping of my sources, because I want to get back to 138 movies.
- The scraping takes forever (much longer than with Kodi 18.8) and I see that now the movie "The Actors" is found all the time, although I don't have this movie at all, and especially not dozens of times.
- After the end of the scraping, I now have 252 movies instead of my 138, including 114 times "The Actors" (see screenshot)
```
These steps for sure are captured at the log. So provide that or don't.

**Comment by DaveTBlake** [comment]
> I have always attached logs to many error messages in the past and still nobody has taken care of it.
> If somebody confirms that the conversion process of myVideosXX.db is also visible in it and the log helps in this case, I might do it again.

@rainman74 several team members have tried to reproduce your issue and been unable to do so.  There is no guaranty that anyone will be able to spend time fixing this issue, or that the log will show us what the issue is. But **without a debug log no one can do anything more**. 

You can also ensure that a log is most useful by turning on debug first (a simple log file contains errors but not detail). Since it is the video db that is being implicated (although your description is hard to follow), it is also worth turning on component specific debug for the database. Do both while using 18.8, and then the full migration to v19 will be fully logged.

**Comment by rainman74** [comment]
Debug.log will come on occasion.

It's not because of MyVideos117.db or MyVideos118.db, because I have just tried
kodi-20200608-00f5af0c-master-armeabi-v7a.apk with MyVideos117.db. Same problem!

Not only the .actors folder will be scraped, but also randomly other subfolders! See screenshots.

https://i.imgur.com/IEF02yN.png
https://i.imgur.com/JVKQ6SJ.png

I use the following exclusions, which worked fine under 18.8
		<excludefromscan action="append">
			<regexp>/Extras/</regexp>
			<regexp>[\\/]Extras[\\/]</regexp>
		</excludefromscan>

**Comment by HitcherUK** [comment]
From the first 2 screenshots how else is the scraper meant to know what the film is if the parent folder to the VIDEO_TS ones are `r-s` and `o-q`?

**Comment by rainman74** [comment]
I have recursive scraping enabled in the Movie Datebase Scraper and it then searches all subfolders until it finds a movie file, i.e. .mp4, .mkv, etc.

And as I said, these VIDEO_TS.IFO files are definitely not in the folder, because these are not DVD folders at all, but regular MPEG4 files with .mkv/.mp4 and an .inf file.

**Comment by rainman74** [comment]
@DaVukovic I have found the cause of the error. It is due to scraping via SMBv1 in Kodi 19, which causes a lot of wrong scapes due to a missing recognition of file names and folders.

But as soon as I scrape in Kodi with SMBv3 from my NAS, everything is correct and all files are scraped correctly.

SMBv1 in Kodi 18 also works fine. Maybe the implementation of SMBv1 was changed in Kodi 19?

What should I do? Should I set the issue here again? You can reproduce it anytime now, if you set max. SMBv1 in Kodi.

**Comment by DaVukovic** [comment]
Maybe I will try, but, tbh, I don't see ANY reason to use SMBv1 anymore. Even Fritzboxes or able to use SBMv2 and higher. We, as TeamKodi, don't even recommend to use SMBv1 at all. So please don't expect much testing, because I also have to change my Linux SMB Server configuration as that doesn't even share SMBv1 anymore. 

> What should I do?

Switch to SMBv2 as the minimum. That's my recommendation. Everything else should be discussed at the forums

**Comment by hegyre** [comment]
Hello guys,
Just saw the discussion and wanted to add my 2 cents.

The issue here seems to be an issue related to SMBv1, which both Kodi v18 and v19 appears to support.

in Kodi v18 it's working fine. But Kodi v19 behaves differently with that same SMBv1.

To me the solution to manually move (manually change setting) to SMBv2 or v3 is not really a solution, because.... Kodi v19 still supports SMBv1. So the bug is not fixed.

If Kodi devs don't want to support SMBv1 in Kodi v19, why not just disallow this protocol in Kodi v19 (and auto-replace setting to SMBv2 during migration to v19), as it seems to be broken ?

It may be a small breaking change for old users but it will also avoid some mess and confusion (and users and devs worktime...)

Cheers

**Comment by hegyre** [comment]
Maybe worth checking the following threads about Kodi v19 as SMB2 becames the default:
- https://forum.kodi.tv/showthread.php?tid=349167
- https://forum.kodi.tv/showthread.php?tid=343069&page=32

Did you try to manually set minimum version to SMBv1 instead of None maybe ? or the reverse...
